tyk-pro/templates/deployment-mdcb.yaml

```diff
@@ -0,0 +1,103 @@
+{{- if .Values.mdcb.enabled -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: mdcb-{{ include "tyk-pro.fullname" . }}
+  labels:
+    app: mdcb-{{ include "tyk-pro.fullname" . }}
+    chart: {{ include "tyk-pro.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.mdcb.replicaCount }}
+  minReadySeconds: 5
+  strategy:
+    # indicate which strategy we want for rolling update
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 2
+      maxUnavailable: 1
+  selector:
+    matchLabels:
+      app: mdcb-{{ include "tyk-pro.fullname" . }}
+      release: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app: mdcb-{{ include "tyk-pro.fullname" . }}
+        release: {{ .Release.Name }}
+    spec:
+{{- if .Values.mdcb.nodeSelector }}
+      nodeSelector:
+{{ toYaml .Values.mdcb.nodeSelector | indent 8 }}
+{{- end }}
+{{- if .Values.mdcb.tolerations }}
+      tolerations:
+{{ toYaml .Values.mdcb.tolerations | indent 8 }}
+{{- end }}
+{{- if .Values.mdcb.affinity }}
+      affinity:
+{{ toYaml .Values.mdcb.affinity | indent 8 }}
+{{- end }}
+      containers:
+      - image: "{{ .Values.mdcb.image.repository }}:{{ .Values.mdcb.image.tag }}"
```

Comment:
The official image is in the private repository, you'll want to specify a configurable [imagePullSecrets](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#create-a-pod-that-uses-your-secret) for this. In fact, this can be useful for all pods in the chart in case someone keeps all their images air-gapped or modified but private. Should be optional of course.